search-TDD/src/main/java/controller/searcher/Searcher.java

```diff
@@ -0,0 +1,18 @@
+package controller.searcher;
+
+import java.util.HashSet;
+
+import controller.DatabaseController;
+import controller.ProgramController;
+
+public interface Searcher {
+    HashSet<String> search(String command);
+
+    default HashSet<String> getWordDocuments(String word) {
```

Comment:
There are to many discussion about using default methods. It s better to avoid this